index.js

```diff
@@ -199,10 +203,63 @@ function session(options){
 
       setcookie(res, name, req.sessionID, secrets[0], cookie.data);
     });
+    
+    var _write = res.write;
+
+    if (saveBeforeLocationHeader) {
+      var _writeHead = res.writeHead;
+      var headWritten = false;
+      res.writeHead = function writeHead(statusCode, reason, obj) {
+        if (headWritten) return;
+        headWritten = true;
 
+        // search for location header only for 3xx status codes
+        if (statusCode < 300 || statusCode >= 400) {
```

Comment:
Won't this not work if the user calls the following, since you are not looking at the already-set response code?
```js
res.statusCode = 301
res.setHeader('Location', '/')
res.end()
```